services/common/src/redux/reducers/authenticationReducer.ts

```diff
@@ -87,8 +87,11 @@ export const getSystemFlag = (state: RootState) => state[AUTHENTICATION].systemF
 export const isAuthenticated = (state: RootState) => state[AUTHENTICATION].isAuthenticated;
 export const getUserAccessData = (state: RootState) => state[AUTHENTICATION].userAccessData;
 export const getUserInfo = (state: RootState) => state[AUTHENTICATION].userInfo;
-export const userHasRole = (state: RootState, role: string) =>
-  state[AUTHENTICATION].userAccessData.includes(USER_ROLES[role] ?? role);
+export const userHasRole = (state: RootState, role: string) => {
+  const userAccessData = state[AUTHENTICATION]?.userAccessData;
+  if (!userAccessData) return false;
```

Comment:
I'm wondering if this is added in order to get the snapshot test to run? Check out MineReportTable.spec.tsx- there the userAccessData is populated via initialState and so we're able to return something other than null or empty div.